sdk/core/core-client-rest/src/sendRequest.ts

```diff
@@ -224,29 +187,57 @@ function processFormData(formData?: FormDataMap) {
 
 /**
  * Prepares the response body
+ * @param response - The received response
+ * @param coerceAs - The type to coerce the body as
  */
-function getResponseBody(response: PipelineResponse): RequestBodyType | undefined {
+async function getResponseBody(
+  response: PipelineResponse,
+  coerceAs: { stream?: boolean } = {}
```

Comment:
why not a single `asStream?: boolean`?

Reply:
Addressed in https://github.com/Azure/azure-sdk-for-js/pull/27000/commits/bf8beb704eace9e1c643949ab758ab27efb38dbc